src/structures/CategoryChannel.js

```diff
@@ -7,6 +7,14 @@ const GuildChannel = require('./GuildChannel');
  * @extends {GuildChannel}
  */
 class CategoryChannel extends GuildChannel {
+  constructor(guild, data) {
```

Comment:
I don't think these changes belong into a PR with the title `fix: Remove setTopic from GuildChannel and apply to TextChannel`.

Also, I do not think that documented methods should be `undefined`, I'd rather have these behave similar to `PartialGroupDMChannel#delete` and `#fetch`: (Make them throw an error instead)
https://github.com/discordjs/discord.js/blob/9ffcd83027f0fc06d69df21475865ad55138de01/src/structures/PartialGroupDMChannel.js#L37-L43
We could additionally to that add a warning to the docstrings of these methods in the `CategoryChannel` class.

Reply:
Wellll, this is consistent with https://github.com/discordjs/discord.js/blob/master/src/structures/NewsChannel.js#L14 where the same thing is done.

Reply:
That is a property, not a method.